### PR TITLE
load organizers for a group

### PR DESCRIPTION
**Pull request by signed**
loading the organizers for a group

### CODE REVIEW

File: softwerkskammer/lib/groupsAndMembers/groupsAndMembersService.js [hunk 1]
```diff
@@ -102,6 +102,18 @@ module.exports = {
...
   },
 
+  getOrganizersOfGroup: function (groupId, callback) {
+    this.getGroupAndMembersForList(groupId, (error, group) => {
+      if (error) {return callback(error);}
+      if (group === undefined) {
+        return callback(null, []);
+      }
+      const groupMembers = group.members || [];
+      const organizers = (groupMembers).filter(member => group.isOrganizer(member.id()));
```
Comment (UrsMetz) [inline softwerkskammer/lib/groupsAndMembers/groupsAndMembersService.js]: I think one can omit the parentheses:
```suggestion
      const organizers = groupMembers.filter(member => group.isOrganizer(member.id()));
```

File: softwerkskammer/test/groupsAndMembers/groupsAndMembersService_test.js [hunk 2]
```diff
@@ -179,6 +204,35 @@ describe('Groups and Members Service (getGroupAndMembersForList)', () => {
...
+    thereIsGroup(new Group({id: groupId, organizers: [organizerId]}));
+    groupHasMembers([organizer, member]);
+    groupsAndMembersService.getOrganizersOfGroup(groupId, (error, organizers) => {
+      expect(organizers).to.have.length(1);
+      expect(organizers[0].id()).to.equal(organizerId);
+      done(error);
+    });
+  });
+});
+
```
Comment (UrsMetz) [inline softwerkskammer/test/groupsAndMembers/groupsAndMembersService_test.js]: Should we maybe have an additional test for the case that there are no organizers in a group? I can also add it later after pulling this request.

File: softwerkskammer/lib/groupsAndMembers/groupsAndMembersService.js [hunk 3]
```diff
@@ -102,6 +102,18 @@ module.exports = {
     });
   },
 
+  getOrganizersOfGroup: function (groupId, callback) {
+    this.getGroupAndMembersForList(groupId, (error, group) => {
+      if (error) {return callback(error);}
+      if (group === undefined) {
```
Comment (UrsMetz) [inline softwerkskammer/lib/groupsAndMembers/groupsAndMembersService.js]: Any reason not to use
```suggestion
      if (!group) {
```

File: softwerkskammer/test/groups/group_object_test.js [hunk 4]
```diff
@@ -192,3 +192,17 @@ describe('returns the meetup :urlname from the given meetup URL', () => {
   });
 
 });
+
+describe('guests can contact the organizers', () => {
+  it('for spam projection purposes is an opt in feature', () => {
```
Comment (UrsMetz) [inline softwerkskammer/test/groups/group_object_test.js]: When i first read the test I was a bit confused what the two different examples meant (now I know :-) ). Should we break this up in two different tests?

**Comment by UrsMetz** [comment]
I have a few points for discussion (see inline comments). I like the DSL-like functions for setting up the stubs more the more I read them :-).

File: softwerkskammer/lib/groupsAndMembers/groupsAndMembersService.js (hunk 3 above)
Comment (signed) [reply softwerkskammer/lib/groupsAndMembers/groupsAndMembersService.js]: Thx, still getting used to javascript :)

File: softwerkskammer/test/groups/group_object_test.js (hunk 4 above)
Comment (signed) [reply softwerkskammer/test/groups/group_object_test.js]: done

File: softwerkskammer/lib/groupsAndMembers/groupsAndMembersService.js (hunk 1 above)
Comment (signed) [reply softwerkskammer/lib/groupsAndMembers/groupsAndMembersService.js]: done

File: softwerkskammer/test/groupsAndMembers/groupsAndMembersService_test.js (hunk 2 above)
Comment (signed) [reply softwerkskammer/test/groupsAndMembers/groupsAndMembersService_test.js]: done

**Comment by UrsMetz** [comment]
@signed thanks for addressing my points :-).
I think the only things left are:
* tests for the added endpoint in the controller (I think the rest of Agora has at least rudimentary tests for these methods)
* UI for organizers to opt-in and out of contact form
* only show link to and accept organizer contact form when the group is configured to do so.

Should we split these tasks up amongst us?

**Comment by signed** [comment]
> I like the DSL-like functions for setting up the stubs more the more I read them :-).

Thats how we get you ;)

**Comment by signed** [comment]
I already have local changes for the UI. But this behaves a bit unexpected. Have to investigate a bit.

* We also need to secure the send mail endpoint to verify that the feature is active for the group

If you have time this week, we can do a remote/in person pair session if you want to?

**Comment by signed** [comment]
I pushed the ui part.
Did you see a check in the submit group part that only allows contact persons to submit changes?

**Comment by UrsMetz** [comment]
@signed I pushed the cleaned up version of what we started yesterday (including the rebase onto origin master).

**Pull request closed without merging**